util.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+)
+
+// SchemaDebug provides a logging interface
+// which is off by defauly but can be activated
+// for debuging purposes
+func SchemaDebug(message string) {
+	debug := false
+	debugEnvVar := os.Getenv("JSON_SCHEMA_DEBUG")
+	if debugEnvVar == "1" {
+		debug = true
+	}
+	if debug {
+		fmt.Printf("%s\n", message)
+	}
+}
```

Comment:
this shouldn't be exported, use `schemaDebug` instead, or even better: `debug(message string, args ...interface{})` and pass that directly to `fmt.Printf`. Callers can refactor this:
```go
SchemaDebug(fmt.Sprintf("message: %v", value))
```

into this:
```go
debug("message: %v", value)
```

Looking at this further, this function is calling out to the operating system _on every call_. Refactoring to this gave a performance bump across the board:

```go
var showDebug = os.Getenv("JSON_SCHEMA_DEBUG") == "1"

// SchemaDebug provides a logging interface
// which is off by defauly but can be activated
// for debuging purposes
func SchemaDebug(message string) {
	if showDebug {
		fmt.Printf("%s\n", message)
	}
}
```